lib/explorer/data_frame.ex

```diff
@@ -1701,6 +1701,75 @@ defmodule Explorer.DataFrame do
     Shared.apply_impl(df, :mutate_with, [df_out, column_pairs])
   end
 
+  @doc """
+  Creates or modifies a single column.
+
+  This is a simplified way to add or modify one column.
+
+  ## Examples
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
+      iex> Explorer.DataFrame.put(df, :b, Explorer.Series.transform(df[:a], fn n -> n * 2 end))
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        a integer [1, 2, 3]
+        b integer [2, 4, 6]
+      >
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
+      iex> Explorer.DataFrame.put(df, :b, [4, 5, 6])
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        a integer [1, 2, 3]
+        b integer [4, 5, 6]
+      >
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
+      iex> Explorer.DataFrame.put(df, :b, 7)
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        a integer [1, 2, 3]
+        b integer [7, 7, 7]
+      >
+
+  ## Grouped examples
+
+  If the dataframe is grouped, then the series or list must be of the
+  same size of each group, or be a scalar value.
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
+      iex> grouped = Explorer.DataFrame.group_by(df, "a")
+      iex> Explorer.DataFrame.put(grouped, :b, [4])
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        Groups: ["a"]
+        a integer [1, 2, 3]
+        b integer [4, 4, 4]
+      >
+
+      iex> df = Explorer.DataFrame.new(a: [1, 2, 3])
+      iex> grouped = Explorer.DataFrame.group_by(df, "a")
+      iex> Explorer.DataFrame.put(grouped, :b, 7)
+      #Explorer.DataFrame<
+        Polars[3 x 2]
+        Groups: ["a"]
+        a integer [1, 2, 3]
+        b integer [7, 7, 7]
+      >
+
+  """
+  @spec put(
+          DataFrame.t(),
+          column_name(),
+          Series.t()
+          | list(Explorer.Backend.Series.valid_types())
+          | Explorer.Backend.Series.valid_types()
+        ) ::
+          DataFrame.t()
+  def put(%DataFrame{} = df, column_name, series) do
+    mutate_with(df, fn _ -> [{column_name, series}] end)
```

Comment:
Shouldn't mutate_with warn if the series is not a LazySeries? I think we would need a new callback for this.

Reply:
BTW, if we introduce a new operation, I would make it so `put` behaves the same regardless if grouped or not. That's because I think having all groups matching is rather unlikely?

Reply:
> Shouldn't mutate_with warn if the series is not a LazySeries? I think we would need a new callback for this.

Now we are implicitly converting a Polars Series to a LazySeries, because a lot of our examples were using series. This is the reason we are not warning.

> BTW, if we introduce a new operation, I would make it so put behaves the same regardless if grouped or not. That's because I think having all groups matching is rather unlikely?

Yeah, having all groups matching is unlikely.
I can add another callback and ignore groups for that. WDYT?